monitoring/notification/dashboard.py

```diff
@@ -0,0 +1,516 @@
+    )],
+    thresholds=[
+        Threshold('green', 1, 0.),
+    ])
+
+bandwidth_rate = Stat(
+    title='Bandwidth Rate',
+    dataSource='${DS_PROMETHEUS}',
+    colorMode='background',
+    noValue='0',
+    reduceCalc='last',
+    format=UNITS.BYTES_SEC,
+    targets=[Target(
+        expr='sum(rate(' + Metrics.NOTIF_SIZE() + '))',
+    )],
+    thresholds=[
+        Threshold('green', 1, 0.),
+    ])
+
+notifications_delivery_rate_per_target = TimeSeries(
+    title='Notification Delivery Rate',
+    description='Number of delivered notifications in a given time',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum by(target) (rate(' + Metrics.DELIVERY_LAG.count(status="success") + '))',
+        legendFormat='{{target}}',
+    )])
+
+notifications_failure_rate_per_target = TimeSeries(
+    title='Failed Notifications Per Target In Time',
+    description='Number of failed notifications in a given time',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum by(target) (rate(' + Metrics.DELIVERY_LAG.count(status="failure") + '))',
+        legendFormat='{{target}}',
+    )])
+
+event_types_processed_rate = TimeSeries(
+    title='Event Types Processed',
+    description='Number of notification events of a given type that where processed',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum by(eventType) (rate(' + Metrics.PROCESSOR_EVENTS() + '))',
+        legendFormat='{{eventType}}',
+    )])
+
+processed_events_in_time = TimeSeries(
+    title='Total Processed Oplog Events',
+    description='Number of processed oplog events by the notification producer',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='hidden',
+    unit=UNITS.OPS_PER_SEC,
+    targets=[Target(
+        expr='sum(rate(' + Metrics.CONSUMED_EVENTS() + '))',
+        legendFormat='{{target}}',
+    )])
+
+bandwith_rate_per_target = TimeSeries(
+    title='Bandwidth',
+    description='Bandwidth used to send notifications to targets in a given time',
+    dataSource='${DS_PROMETHEUS}',
+    gradientMode='opacity',
+    lineInterpolation='smooth',
+    fillOpacity=10,
+    spanNulls=True,
+    legendDisplayMode='table',
+    legendPlacement='right',
+    unit=UNITS.BYTES,
```

Comment:
How about adding the time unit in the y axis here? `UNIT.BYTES_SEC`

I am not sure, when looking at the graph, if this is a sum of all data processed by a unit of time, or per second (without looking at the code, as a simple user). If you agree having a 1 second resolution might help reading.

Reply:
good point